### PR TITLE
feat: add EIP-6963 support

### PR DESCRIPTION
**Pull request by wa0x6e**
When multiple wallets are injected, `window['ethereum'].request()` will call the latest injected wallets.

This PR will migrate the injected wallets detection logic to use EIP-6963, and return the correct wallet chosen by the user.

Will also return the list of injected wallets from EIP-6863, instead of hardcoding our own list

### Test

1. `yarn build`
2. `yarn link`
3. Go to your snapshot or sx repo
4. `yarn link "@snapshot-labs/lock"`
5. `yarn dev`
6. Confirm that it uses and remember your chosen wallet

### CODE REVIEW

**Comment by bonustrack** [comment]
Can we support EIP-6963 by the way?

**Comment by wa0x6e** [comment]
> Can we support EIP-6963 by the way?

Wasn't there plan to deprecate this package, and use wagmi instead ?

**Comment by bonustrack** [comment]
Yes but there was something blocking, I think wagmi was only supported for React

**Comment by ChaituVR** [comment]
[EIP-6963 ](https://eip6963.org/)is the best way to move forward

**Comment by wa0x6e** [comment]
After reading the code from sx again, this PR does not make a lot of sense:

SX is using:

```
  injected: {
    id: 'injected',
    name: 'MetaMask',
    type: 'injected',
    root: 'ethereum',
    icon: metamaskIcon
  },
```

which assume that the `injected` wallet is always metamask.

This is not what lock is returning before, nor after this PR.

The injected connector is just returning any injected wallet, not specifically metamask.

After this PR, it will prioritize metamask, but can still return anything else if metamask is not installed

**Comment by wa0x6e** [comment]
Will try EIP-6963 instead, seems like `window.ethereum` has always been unable to accept more than one provider.

**Comment by bonustrack** [comment]
I tried yesterday but couldn't reproduce my issue locally (without this PR), only in prod it trigger me Rabby, 6963 would be great

**Review by bonustrack — COMMENTED** [review]
Works for me, the only issue is that it still show "MetaMask" in the modal for login when Rabby is the used as main wallet
image

File: connectors/injected.ts [hunk 1]
```diff
@@ -1,26 +1,84 @@
 import LockConnector from '../src/connector';
 
+declare global {
+  interface WindowEventMap {
+      "eip6963:announceProvider": EIP6963AnnounceProviderEvent;
```
Comment (bonustrack) [inline connectors/injected.ts:5]: intend is wrong in this file, should be 2 space but its 4 now

**Comment by wa0x6e** [comment]
> Works for me, the only issue is that it still show "MetaMask" in the modal for login when Rabby is the used as main wallet image

Yes, it was mentioned in my comment above, metamask is hardcoded as the injected wallet provider on the UI side.

**Comment by bonustrack** [comment]
Hum, doesn't seem to work for me, when I lock Metamask and refresh the page it will still trigger Metamask for login

**Comment by wa0x6e** [comment]
Yes, that's the title of the pr

**Comment by wa0x6e** [comment]
I will refactor the whole thing to fully support and take advantage of the EIP-6963 standard, instead of just fixing the discussed issue

**Comment by bonustrack** [comment]
@wa0x6e We decided to move this to next cycle for now to focus on more user facing feature

**Pull request closed without merging**